NitroxClient/Communication/Packets/Processors/KnownTechEntryAddProcessor.cs

```diff
@@ -1,37 +1,31 @@
-﻿using System;
 using NitroxClient.Communication.Abstract;
 using NitroxClient.Communication.Packets.Processors.Abstract;
 using NitroxModel.Packets;
 using NitroxModel_Subnautica.DataStructures;
 
-namespace NitroxClient.Communication.Packets.Processors
+namespace NitroxClient.Communication.Packets.Processors;
+
+public class KnownTechEntryProcessorAdd : ClientPacketProcessor<KnownTechEntryAdd>
 {
-    public class KnownTechEntryProcessorAdd : ClientPacketProcessor<KnownTechEntryAdd>
-    {
-        private readonly IPacketSender packetSender;
+    private readonly IPacketSender packetSender;
 
-        public KnownTechEntryProcessorAdd(IPacketSender packetSender)
-        {
-            this.packetSender = packetSender;
-        }
+    public KnownTechEntryProcessorAdd(IPacketSender packetSender)
+    {
+        this.packetSender = packetSender;
+    }
 
-        public override void Process(KnownTechEntryAdd packet)
+    public override void Process(KnownTechEntryAdd packet)
+    {
+        using (packetSender.Suppress<KnownTechEntryAdd>())
         {
-            using (packetSender.Suppress<KnownTechEntryAdd>())
+            switch (packet.Category)
             {
-                switch (packet.Category)
-                {
-                    case KnownTechEntryAdd.EntryCategory.KNOWN:
-                        KnownTech.Add(packet.TechType.ToUnity(), packet.Verbose);
-                        break;
-                    case KnownTechEntryAdd.EntryCategory.ANALYZED:
-                        KnownTech.Analyze(packet.TechType.ToUnity(), packet.Verbose);
-                        break;
-                    default:
-                        string categoryName = Enum.GetName(typeof(KnownTechEntryAdd.EntryCategory), packet.Category);
-                        Log.Error($"Received an unknown category type for {nameof(KnownTechEntryAdd)} packet: {categoryName}");
-                        break;
-                }
```

Comment:
As this field is strictly filled with one of each enum's value, there's no possible way of getting something else than KNOWN or ANALYZED from it, except if the serializer messes it up or if the original packet was sent from a modified client/different version. If we still want this default behaviour, it would be more relevant on server-side.

Reply:
Typically a default case with an enum protects against adding a new enum value without modifying the switch statement.  That would be a silent failure.